src/App.tsx

```diff
@@ -1,14 +1,61 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './TodoStatus';
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [isTodosLoading, setIsTodosLoading] = useState(false);
+  const [filterByStatus, setFilterByStatus] = useState<TodoStatus>(
+    TodoStatus.ALL,
+  );
+  const [searchByName, setSearchByName] = useState<string>('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [todosData, setTodosData] = useState<Todo[]>([]);
+
+  useEffect(() => {
+    setIsTodosLoading(true);
+    getTodos().then(todos => {
+      // eslint-disable-next-line no-console
+      setTodosData(todos);
+      setTodoList(todos);
+      setIsTodosLoading(false);
+    });
+  }, []);
+
+  useEffect(() => {
+    let filteredTodos = [...todosData];
```

Comment:
You are doing an extra copy here

```suggestion
    let filteredTodos = todosData;
```